build_tools/cmake/iree_hal_cts_test_suite.cmake

```diff
@@ -208,3 +208,13 @@ function(iree_hal_cts_test_suite)
     )
   endforeach()
 endfunction()
+
+function(iree_hal_cts_semaphore_failure_tests_filter_list OUT_VAR)
+  unset(FILTER_LIST)
+  string(APPEND FILTER_LIST "SemaphoreTest.WaitThenFail:")
+  string(APPEND FILTER_LIST "SemaphoreTest.FailThenWait:")
+  string(APPEND FILTER_LIST "SemaphoreTest.MultiWaitThenFail:")
+  string(APPEND FILTER_LIST "SemaphoreTest.DeviceMultiWaitThenFail:")
+  string(APPEND FILTER_LIST "SemaphoreSubmissionTest.PropagateFailSignal:")
+  set(${OUT_VAR} "${FILTER_LIST}" PARENT_SCOPE)
+endfunction()
```

Comment:
This specific list should not be tracked in this file - it should be tracked where the test suite is instantiated in each `cts/CMakeLists.txt` file. Build system functions should be generic - let the leaf build graph nodes make local decisions.

General support for filtering individual test cases could be added on the `iree_hal_cts_test_suite` function with an option like the existing `EXCLUDED_TESTS`, e.g. `EXCLUDED_TEST_CASES` that would add the `--gtest_filter` arg.

Reply:
I removed the function.